compiler/noirc_errors/src/debug_info.rs

```diff
@@ -18,6 +23,8 @@ pub struct DebugInfo {
     /// that they should be serialized to/from strings.
     #[serde_as(as = "BTreeMap<DisplayFromStr, _>")]
     pub locations: BTreeMap<OpcodeLocation, Vec<Location>>,
+    pub variables: HashMap<u32, (String, u32)>, // var_id => (name, type_id)
+    pub types: HashMap<u32, PrintableType>,     // type_id => printable type
```

Comment:
Note to self to look at how this acts when we're not in a debug context. Should we make this optional?